### PR TITLE
AuthenticatedUserDAO

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added ServiceProvider class and relationship to User class
2. Implemented AuthenticatedUserDAO

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/auth/AuthenticatedUserDAO.java [hunk 1]
```diff
@@ -10,7 +16,12 @@
 import foam.nanos.auth.AuthService;
 import foam.nanos.auth.User;
 import java.security.AccessControlException;
+
+import foam.util.SafetyUtil;
 import net.nanopay.model.Account;
+import org.apache.http.util.TextUtils;
```
Comment (adamvy) [inline src/foam/nanos/auth/AuthenticatedUserDAO.java]: Are you actually using this import anywhere?

File: src/foam/nanos/auth/AuthenticatedUserDAO.java [hunk 2]
```diff
@@ -10,7 +16,12 @@
 import foam.nanos.auth.AuthService;
 import foam.nanos.auth.User;
 import java.security.AccessControlException;
+
+import foam.util.SafetyUtil;
 import net.nanopay.model.Account;
```
Comment (adamvy) [inline src/foam/nanos/auth/AuthenticatedUserDAO.java]: We need to something about the fact that this class in FOAM2 is referencing a class in nanopay.  Doesn't have to be done in the PR but we need a plan for it soon, right now building nanos outside of nanopay is broken.

Comment (kirkeaton) [reply src/foam/nanos/auth/AuthenticatedUserDAO.java]: It's not actually being used in this class so I can just remove it.

Comment (adamvy) [reply src/foam/nanos/auth/AuthenticatedUserDAO.java]: Did you reply to the right comment?  It looks like Account is used in many places in this class.

Comment (kirkeaton) [reply src/foam/nanos/auth/AuthenticatedUserDAO.java]: I did reply to the right comment. It's not being used anywhere here.

File: src/foam/nanos/auth/AuthenticatedUserDAO.java (hunk 1 above)
Comment (kirkeaton) [reply src/foam/nanos/auth/AuthenticatedUserDAO.java]: No I just cleaned up the imports.

File: src/foam/nanos/auth/AuthenticatedUserDAO.java (hunk 2 above)
Comment (adamvy) [reply src/foam/nanos/auth/AuthenticatedUserDAO.java]: Oh I see, didn't read your changes closely enough, you removed all the references to Account.

**Review by adamvy — APPROVED** [review]

**Pull request merged**